include/linux/blkdev_compat.h

```diff
@@ -301,68 +302,113 @@ bio_set_flags_failfast(struct block_device *bdev, int *flags)
  * allow richer semantics to be expressed to the block layer.  It is
  * the block layers responsibility to choose the correct way to
  * implement these semantics.
- *
- * The existence of these flags implies that REQ_FLUSH an REQ_FUA are
- * defined.  Thus we can safely define VDEV_REQ_FLUSH and VDEV_REQ_FUA
- * compatibility macros.
- *
- * Linux 4.8 renamed the REQ_FLUSH to REQ_PREFLUSH but there was no
- * functional change in behavior.
  */
 #ifdef WRITE_FLUSH_FUA
-
 #define	VDEV_WRITE_FLUSH_FUA		WRITE_FLUSH_FUA
-#ifdef REQ_PREFLUSH
-#define	VDEV_REQ_FLUSH			REQ_PREFLUSH
-#define	VDEV_REQ_FUA			REQ_FUA
 #else
-#define	VDEV_REQ_FLUSH			REQ_FLUSH
-#define	VDEV_REQ_FUA			REQ_FUA
+#define	VDEV_WRITE_FLUSH_FUA		WRITE_BARRIER
 #endif
 
+/*
+ * 4.8 - 4.x API,
+ *   REQ_OP_FLUSH
+ *
+ * 4.8-rc0 - 4.8-rc1,
+ *   REQ_PREFLUSH
+ *
+ * 2.6.36 - 4.7 API,
+ *   REQ_FLUSH
+ *
+ * 2.6.x - 2.6.35 API,
+ *   HAVE_BIO_RW_BARRIER
+ */
+static inline boolean_t
+bio_is_flush(struct bio *bio)
+{
+#if defined(HAVE_REQ_OP_FLUSH)
+	return (bio_op(bio) == REQ_OP_FLUSH);
+#elif defined(REQ_PREFLUSH)
+	return (bio->bi_rw & REQ_PREFLUSH);
```

Comment:
I'm certain REQ_FLUSH turned in to REQ_PREFLUSH and then that changed in to REQ_OP_FLUSH.  See commit torvalds/linux@3a5e02ced11e22ecd9da3d6710afe15bcfee1d10